wp-cli/vip-migrations.php

```diff
@@ -4,6 +4,113 @@
 
 class VIP_Go_Migrations_Command extends WPCOM_VIP_CLI_Command {
 
+	/**
+	 * Run cleanup on the current site database.
+	 *
+	 * [--network]
+	 * : Cleanup all sites on the network
+	 *
+	 * [--dry-run]
+	 * : Show changes without updating
+	 *
+	 * [--yes]
+	 * : Skip the confirmation prompt
+	 */
+	function cleanup( $args, $assoc_args ) {
+		global $wpdb;
+		
+		$dry_run = Utils\get_flag_value( $assoc_args, 'dry-run' );
+		if ( $dry_run ) {
+			WP_CLI::log( 'Performing a dry run, with no database modification.' );
+		} else {
+			$env = defined( 'VIP_GO_ENV' ) ? VIP_GO_ENV : 'unknown';
+			WP_CLI::confirm( sprintf( 'Are you sure you want to run cleanup on the %s environment?', $env ) , $assoc_args );
+		}
+
+		$network = Utils\get_flag_value( $assoc_args, 'network' );
+		if ( $network && ! is_multisite() ) {
+			WP_CLI::warning( 'This is not a multisite install. Proceeding as single site.' );
+			$network = false;
+		}
+		
+		if ( $network ) {
+			$iterator_args = array(
+				'table' => $wpdb->blogs,
+				'where' => array( 'spam' => 0, 'deleted' => 0, 'archived' => 0 ),
+			);
+			$it = new \WP_CLI\Iterators\Table( $iterator_args );
+			foreach ( $it as $blog ) {
+				$url = $blog->domain . $blog->path;
+				$cmd = "--url={$url} vip migration cleanup";
+
+				if ( $dry_run ) {
+					$cmd .= ' --dry-run';
+				}
+
+				WP_CLI::line();
+				WP_CLI::line( 'Cleaning: ' . $url );
+				WP_CLI::runcommand( $cmd );
+			}
+
+			WP_CLI::line( 'Flushing object cache' );
+			if ( ! $dry_run ) {
+				wp_cache_flush();
+			}
+
+			return;
+		}
+
+		// Cleanup options
+		$options = [
+			'jetpack_options',
+			'jetpack_private_options',
+			'vaultpress',
+			'wordpress_api_key',
+		];
+
+		foreach ( $options as $option ) {
+			WP_CLI::line( 'Deleting option: ' . $option );
+
+			if ( ! $dry_run ) {
+				delete_option( $option );
+			}
+		}
+
+		// Cleanup transients
+		$transients = $wpdb->get_col(
+			"SELECT option_name FROM $wpdb->options
+			WHERE option_name LIKE '\_transient\_%'
+			OR option_name LIKE '\_site\_transient\_%'"
+		);
+
+		WP_CLI::line( 'Deleting transients: ' . implode( ', ', $transients ) );
+
+		if ( ! $dry_run ) {
+			$wpdb->query(
+				"DELETE FROM $wpdb->options
+				WHERE option_name LIKE '\_transient\_%'
+				OR option_name LIKE '\_site\_transient\_%'"
+			);
+		}
+
+		do_action( 'vip_go_migration_cleanup', $dry_run );
+
+		// Flush the cache unless we're cleaning up the entire network
+		if ( ! $network ) {
```

Comment:
Small note - this conditional may not be necessary due to the prior `if ( $network ) { ... return; }`.

Reply:
We call the same command again, but with a `--url` specified. The goal is to avoid flushing the cache N times for multisites.

Reply:
Although now I'm realizing this will always be false 🤔